rita_exit/src/database/database_tools.rs

```diff
@@ -348,11 +410,366 @@ pub fn create_or_update_user_record(
 
         let new_ip = get_next_client_ip(conn)?;
 
-        let c = client_to_new_db_client(client, new_ip, user_country);
+        let internet_ip = get_client_subnet(subnet, subnet_entry, conn)?;
+
+        let c = client_to_new_db_client(client, new_ip, user_country, internet_ip);
 
         info!("Inserting new client {}", client.global.wg_public_key);
         diesel::insert_into(clients).values(&c).execute(conn)?;
 
         Ok(c)
     }
 }
+
+/// This function creates an entry for the given subnet in the assgined_ips table if doesnt exist
+fn initialize_subnet_datastore(
+    sub: IpNetwork,
+    conn: &PgConnection,
+) -> Result<models::AssignedIps, RitaExitError> {
+    use self::schema::assigned_ips::dsl::{assigned_ips, subnet};
+    let filtered_list = assigned_ips.filter(subnet.eq(sub.to_string()));
+    match filtered_list.load::<models::AssignedIps>(conn) {
+        Err(_) => {
+            let record = AssignedIps {
+                subnet: sub.to_string(),
+                available_subnets: "".to_string(),
+                iterative_index: 0,
+            };
+            diesel::insert_into(assigned_ips)
+                .values(&record)
+                .execute(conn)?;
+            Ok(record)
+        }
+        Ok(mut a) => {
+            if a.len() > 1 {
+                error!("More than one entry for singular subnet in database, please fix");
```

Comment:
if this is a failure condition should we maybe panic instead of just logging?

Reply:
This wont break the functinoality of the program. There should be only one entry, but there will be two entries if there is an 
unlikely race condition, the code will still work as normal, except it will miss some available subnets. Should I still panic in this case?